eventlog/service.go

```diff
@@ -119,28 +111,112 @@ func (s *Service) GetEvent(req *GetEventRequest) (*GetEventResponse, error) {
 	}, nil
 }
 
+// This should be a const, but we want to be able to hack it from tests.
+var searchMax = 10000
+
+// Search will search the event log for matching entries.
+func (s *Service) Search(req *SearchRequest) (*SearchResponse, error) {
+	if req.ID.IsNull() {
+		return nil, errors.New("skipchain ID required")
+	}
+
+	if req.To == 0 {
+		req.To = time.Now().UnixNano()
+	}
+
+	v := s.omni.GetCollectionView(req.ID)
+	el := &eventLog{ID: theEventLog.Slice(), v: v}
+
+	id, b, err := el.getLatestBucket()
+	if err == errIndexMissing {
+		// There are no events yet on this chain, so return no results.
+		return &SearchResponse{}, nil
+	}
+	if err != nil {
+		return nil, err
+	}
+	// bEnd is normally updated from the last bucket's start. For the latest
+	// bucket, bEnd is now.
+	bEnd := time.Now().UnixNano()
```

Comment:
if I understood correctly, this might not be the case if you submit an even that has a timestamp after `time.Now()` and then do a search request immediately after the event is recorded.

Reply:
I see what you are saying, but it seems too uncommon, let's keep this as simple as possible until it proves to be a problem.